src/charm.py

```diff
@@ -116,6 +116,12 @@ def _on_collect_unit_status(self, event: CollectStatusEvent):
             event.add_status(WaitingStatus("Waiting for N2 information"))
             logger.info("Waiting for N2 information")
             return
+        if not self._core_gnb_requirer.tac or not (plmns := self._core_gnb_requirer.plmns):
+            event.add_status(WaitingStatus("Waiting for TAC and PLMNs configuration"))
+            return
+        if not self._is_valid_plmn(plmns[0]):
+            event.add_status(BlockedStatus("Invalid PLMNs configuration"))
+            return
```

Comment:
I think we could be more specific here. This check is really just for the `SD`. I'd either say that the `SD` is missing  from the PLMN config (in which case method name should be updated too) or check all the params and list those that are missing (i.e. "Invalid PLMN configuration. Missing values: blablabla"). 
Personally I'd go with the first option, because we already have data validation in the lib.